### ISSUE TITLE
[pickers] DatePicker cannot select date with VoiceOver and Full Keyboard Access in Pass Through mode

### ISSUE THREAD

**Issue by petercutting** [mui/mui-x#20632]
### Steps to reproduce

Steps:
using iOS phone with Bluetooth keyboard
1. surf to https://mui.com/x/react-date-pickers/date-picker/ (I used safari)
2. find any date picker showing month as a matrix of days
3. in iOS accessibility settings turn on Full Keyboard Access . note Pass Through mode shortcut
4. turn on VoiceOver
5. date matrix can be navigated with arrow keys and date can be selected with space/return. Great
6. turn on Pass Through mode using the shortcut (default Ctrl Opt Cmd P)
7. navigation still works with arrow keys but space/return does NOT select a date

for some reason my Cordova Mui v5 app does not work properly with FKA On and Pass Through mode Off - for example onKeyDown events do not fire. This is a shame as I would prefer that it just worked. Its unclear why it does not work

Pass Through mode On seemed to fix everything until I discovered that I could not select a date using the BT keyboard

### Your environment

<details>
  <summary><code>npx @mui/envinfo</code></summary>

```
  Don't forget to mention which browser you used.
  Output from `npx @mui/envinfo` goes here.
```
</details>

**Search keywords**: VoiceOver Full Keyboard Access Pass Through mode

**Comment by petercutting** [comment]
I have also noticed a similar VoiceOver problem running my app on MacOS, Chrome and Safari. With VoiceOver Off, I can use the date picker OK with the keyboard:-
- Tab into the dates matrix OK
- arrow keys to move between dates OK
- space/return to select date OK

However if I turn On VoiceOver then I can no longer select a date:-
- Tab into the dates matrix OK
- arrow keys to move between dates OK but focus indicator is now different:- small ring
- space/return to select date NOTHING HAPPENS

**Comment by michelengelen** [comment]
Hey @petercutting,
sry for answering so late ... we had limited capacity during the holiday season.

I will look into this once i caught up with everything! 👍🏼

**Comment by michelengelen** [comment]
Hey @petercutting I have opened a PR to fix this. It is a bit of a shot into the dark, since I could not fully test this, but during my research I read a few times that pass through sometimes prevents the event to be picked up correctly from the React event system.
